apps/condo/domains/resident/access/Resident.js

```diff
@@ -18,8 +20,28 @@ async function canReadResidents ({ authentication: { item: user } }) {
 async function canManageResidents ({ authentication: { item: user }, originalInput, operation, itemId, context }) {
     if (!user) return false
     if (user.isAdmin) return true
-    if (operation === 'create' && user.type === RESIDENT) {
-        return true
+    if (user.type === RESIDENT) {
+        // Only soft-delete is allowed for current resident
+        if (operation === 'update') {
+            if (!itemId) {
+                return false
+            }
+            const [resident] = await Resident.getAll(context, { id: itemId })
+            if (!resident) {
+                return false
+            }
+            if (user.id !== resident.user.id) {
+                return false
+            }
+            if (
+                originalInput.deletedAt !== null &&
```

Comment:
Maybe there is a more simple way to check - without checking Object.keys(originalInput).length === 3
At least we can define some function to check if update action was in fact softDelete to use this later

Reply:
Extracted in `9eb94e62`